Doc/whatsnew/3.12.rst

```diff
@@ -754,6 +754,15 @@ Changes in the Python API
   around process-global resources, which are best managed from the main interpreter.
   (Contributed by Dong-hee Na in :gh:`99127`.)
 
+* The undocumented locking behavior of :func:`~functools.cached_property`
+  is removed, because it locked across all instances of the class, leading to high
+  lock contention. This means that a cached property getter function could now
+  occasionally run more than once for a single instance, if two threads race. For
+  most simple cached properties (e.g. those that are idempotent and simply
```

Comment:
The word "occasionally" is not necessary -- we already have the "could" and "if".